src/levanter/compat/torch_serialization.py

```diff
@@ -120,45 +121,37 @@ def jax_tree_to_state_dict(tree: PyTree, prefix: Optional[str] = None) -> StateD
 
 
 def default_eqx_module_from_state_dict(mod: Mod, state_dict: StateDict, prefix: Optional[str] = None) -> Mod:
-    key_map = None
-    if hasattr(mod, "_state_dict_key_map"):
-        key_map = mod._state_dict_key_map()
-
-    old_values, mod_state = mod.tree_flatten()
-    dyn_keys = mod_state[0]
-
-    new_values = []
-    for k, old in zip(dyn_keys, old_values):
-        if key_map is not None and k in key_map:
-            k = key_map[k]
+    key_map = getattr(mod, "_state_dict_key_map", lambda: {})()
+    names = []
+    values = []
+    for field in fields(mod):
+        if field.metadata.get("static", False):
+            continue
+        key = key_map.get(field.name, field.name)
+        value = getattr(mod, field.name)
         # TODO: might want to add a flag that allows missing keys?
-        new_values.append(jax_tree_from_state_dict(old, state_dict, apply_prefix(prefix, k)))
+        new = jax_tree_from_state_dict(value, state_dict, apply_prefix(prefix, key))
+        names.append(field.name)
+        values.append(new)
+    return eqx.tree_at(lambda m: [getattr(m, name) for name in names], mod, values)
 
-    return mod.tree_unflatten(mod_state, new_values)
```

Comment:
Changes in this file are because `equinox.Module.tree_{flatten,unflatten}` have been removed. (They were actually always private API, and have now been properly tucked away.)

Reply:
do i remember right you exposed key paths recently? That's really what this whole thing is trying to do anyway.

Reply:
Yup!